Documentation/performance/JitOptimizerTodoAssessment.md

```diff
@@ -0,0 +1,133 @@
+Optimizer Codebase Status/Investments
+=====================================
+
+There are a number of areas in the optimizer that we know we would invest in
+improving if resources were unlimited.  This document lists them and some
+thoughts about their current state and prioritization, in an effort to capture
+the thinking about them that comes up in planning discussions.
+
+
+Improved Struct Handling
+------------------------
+
+This is an area that has received recent attention, with the first-class structs
+work and the struct promotion improvements that went in for `Span<T>`.  Work here
+is expected to continue and can happen incrementally.  Possible next steps:
+
+ - Struct promotion stress mode (test mode to improve robustness/reliability)
+ - Promotion of more structs; relax limits on e.g. field count (should generally
+   help performance-sensitive code where structs are increasingly used to avoid
+   heap allocations)
+ - Improve handling of System V struct passing (I think we currently insert
+   some unnecessary round-trips through memory at call boundaries due to
+   internal representation issues)
+ - Implicit byref parameter promotion w/o shadow copy
+
+We don't have specific benchmarks that we know would jump in response to any of
+these.  May well be able to find some with some looking, though this may be an
+area where current performance-sensitive code avoids structs.
+
+
+Exception handling
+------------------
+
+This is increasingly important as C# language constructs like async/await and
+certain `foreach` incantations are implemented with EH constructs, making them
+difficult to avoid at source level.  The recent work on finally cloning, empty
+finally removal, and empty try removal targeted this.  Writethrough is another
+key optimization here, and we are actively pursuing it.  Other things we've
+discussed include inlining methods with EH and computing funclet callee-save
+register usage independently of main function callee-save register usage, but
+I don't think we have any particular data pointing to either as a high priority.
```

Comment:
While EH Write Thru is not strictly an optimization, the lack of it is an inhibitor to improve performance of EH-intensive code (see https://github.com/dotnet/coreclr/blob/master/Documentation/design-docs/eh-writethru.md, and I also submitted a PR with a suggested alternate approach).